### PR TITLE
refactor(core/jquery-enhanced): migrate $renameElement to vanilla js

### PR DESCRIPTION
**Pull request by sidvishnoi**
As a part of https://github.com/w3c/respec/issues/1699

### CODE REVIEW

File: src/core/utils.js [hunk 1]
```diff
@@ -737,3 +737,35 @@ export function getLinkTargets(elem) {
...
+  const newElement = elem.ownerDocument.createElement(newName);
+
+  // copy attributes
+  for (let i = 0, n = elem.attributes.length; i < n; i++) {
+    const attr = elem.attributes[i];
+    try {
+      newElement.setAttributeNS(attr.namespaceURI, attr.name, attr.value);
+    } catch (err) {
+      let msg = "Your HTML markup is malformed. Error in: \n";
+      msg += "```HTML\n" + this.outerHTML + "\n```";
```
Comment (sidvishnoi) [inline src/core/utils.js]: oops. s/this/elem/

Comment (marcoscaceres) [reply src/core/utils.js]: oops. Yeah, please fix that.

File: src/core/utils.js [hunk 2]
```diff
@@ -737,3 +737,35 @@ export function getLinkTargets(elem) {
...
+ * @returns {Element} new renamed element
+ */
+export function renameElement(elem, newName) {
+  const newElement = elem.ownerDocument.createElement(newName);
+
+  // copy attributes
+  for (let i = 0, n = elem.attributes.length; i < n; i++) {
+    const attr = elem.attributes[i];
+    try {
+      newElement.setAttributeNS(attr.namespaceURI, attr.name, attr.value);
```
Comment (marcoscaceres) [inline src/core/utils.js]: Nit: I know this is copy/paste... but we can just jus `setAttribute`. We don't support XHTML anymore.

Comment (sidvishnoi) [reply src/core/utils.js]: Removing this exception handling as we have `renameElement` as an internal function, and we rename only specific elements. So, we never get this exception :crossed_fingers:.

**Pull request merged**